File: Factorial/factorial.java
import java.util.Scanner;
/*This a example of Factorization*/
public class Main {
    public static long factorial(int n){
        long accumulator = 1;

        for(; n > 0; n--){
            accumulator *= n;
        }

        return accumulator;
    }

    public static boolean isInteger(String text){
        try{
            Integer.parseInt(text);
            return true;
        }catch(Exception e){
            return false;
        }
    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter a number to factorialize: ");
        String input = scanner.nextLine();

        if(isInteger(input)){
            try{
                int num = Integer.parseInt(input);
                System.out.println("The factorial of " + num + " is " + factorial(num));
            }catch(NumberFormatException e){
                System.out.println("What happened there ?");
            }
        }else {
            System.out.println("The input was not a number");
        }
    }
}

File: Factorial/Factorial.py
def factorial(n):
    fact = 1
    for i in range(1, n+1):
        fact *= i
    return fact
 
def factorial_recursion(n):
    if (n == 0):
        return 1
    else:
        return n * factorial_recursion(n-1)

number = int(input("Enter a number: "))
print("Fctorial using Recursion: ", factorial_recursion(number))
print("Fctorial using Iterative approach: ", factorial(number))
